### PR TITLE
Iir coefficients.py

### PR DESCRIPTION
**Pull request by nkrackow**
This PR adds python utilities to Thermostat. One script is for setting up an output channel with input matrix, iir, target temperature etc. Another is for plotting the iir response.

### CODE REVIEW

**Comment by nkrackow** [comment]
Note that this is black formatted. Should I switch back to pep8 like in Stabilizer?

File: py/thermostat/configure_output_ch.py [hunk 1]
```diff
@@ -0,0 +1,419 @@
+#!/usr/bin/python3
+"""
+Adapted from Stabilizer iir_coefficients.py.
```
Comment (jordens) [inline py/thermostat/configure_output_ch.py:3]: Why not `from stabilizer import ...`?

File: py/thermostat/plot_iir_frequency_response.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/python3
+"""
+Small tool to show the frequency response of biquad IIR filters.
```
Comment (jordens) [inline py/thermostat/plot_iir_frequency_response.py:3]: Just use the one in stabilizer.
If you copy things, you have to carry the burden of maintaining that code, checking divergence, porting changes/fixes/improvements in both ways. It's not a good idea if you can just use existing code.

Comment (jordens) [reply py/thermostat/plot_iir_frequency_response.py:3]: And obviously nobody will review duplicate code.

File: py/thermostat/configure_output_ch.py (hunk 1 above)
Comment (nkrackow) [reply py/thermostat/configure_output_ch.py:3]: Ok.

**Comment by jordens** [comment]
> Note that this is black formatted. Should I switch back to pep8 like in Stabilizer?

Black is fine

File: py/setup.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+setup(
+    name="thermostat",
+    packages=find_packages(),
+    # Keep versions in Cargo.toml and py/setup.py synchronized.
+    version="0.1.0",
+    description="Thermostat Utilities",
+    author="QUARTIQ GmbH",
+    license="MIT",
+    install_requires=[
+        "miniconf-mqtt@git+https://github.com/quartiq/miniconf@develop#subdirectory=py/miniconf-mqtt",
```
Comment (jordens) [inline py/setup.py:16]: And stabilizer

Comment (nkrackow) [reply py/setup.py:16]: ah yes

File: py/thermostat/configure_output_ch.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        "--y-offset",
+        type=float,
+        default=0,
+        help="The channel output offset (%(default)s V)",
+    )
+    parser.add_argument(
+        "--input-weights",
+        type=float,
+        default=[1, 0, 0, 0, 0, 0, 0, 0],
+        help="Input channel weights (%(default)s V)",
```
Comment (jordens) [inline py/thermostat/configure_output_ch.py:90]: Those are not voltages.

File: py/thermostat/configure_output_ch.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        "--y-max",
+        type=float,
+        default=1,
+        help="The channel maximum output (%(default)s V)",
+    )
+    parser.add_argument(
+        "--y-offset",
+        type=float,
+        default=0,
+        help="The channel output offset (%(default)s V)",
```
Comment (jordens) [inline py/thermostat/configure_output_ch.py:84]: All not `V`, right?

File: py/thermostat/configure_output_ch.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        prefix = devices[0]
+        logger.info("Automatically using detected device prefix: %s", prefix)
+
+    async def configure():
+        logger.info("Connecting to broker")
+        interface = await miniconf.Miniconf.create(prefix, args.broker)
+
+        # Set the filter coefficients.
+        await interface.command(
+            f"output_channel/{args.channel}",
```
Comment (jordens) [inline py/thermostat/configure_output_ch.py:156]: Ah. Does that have to be a `MiniconfAtomic`? IMO having `iir` as `MiniconfAtomic` is sufficient. The rest doesn't need to be synchronized.

Comment (jordens) [reply py/thermostat/configure_output_ch.py:156]: https://github.com/quartiq/thermostat-eem/blob/d8aef7332822e2c04ced8d273dbbede92d5501c3/src/output_channel.rs#L8

File: py/thermostat/configure_output_ch.py (hunk 5 above)
Comment (nkrackow) [reply py/thermostat/configure_output_ch.py:84]: thanks

**Review by jordens — APPROVED** [review]
Apart from the units LGTM

**Comment by nkrackow** [comment]
> Apart from the units LGTM

Ok, I'll fix that soon and switch to Miniconf for OutputChannel and then merge.

File: py/thermostat/configure_output_ch.py (hunk 6 above)
Comment (nkrackow) [reply py/thermostat/configure_output_ch.py:156]: Ah I remember why I did it this way. Because the weights then become an individual path each.

**Pull request merged**